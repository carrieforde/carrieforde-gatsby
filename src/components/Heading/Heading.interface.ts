import { HeadingLevel } from 'entities/heading-level.type';

export interface HeadingProps {
  level: HeadingLevel;
  className?: string;
}
