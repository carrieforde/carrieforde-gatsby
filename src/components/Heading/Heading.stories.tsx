import React from 'react';
import Heading from './Heading';
import { HeadingProps } from './Heading.interface';
import { Story, Meta } from '@storybook/react';

export default {
  title: 'Heading',
  component: Heading,
} as Meta;

const Template: Story<HeadingProps> = (args) => <Heading {...args} />;

export const H1 = Template.bind({});
export const H2 = Template.bind({});
export const H3 = Template.bind({});
export const H4 = Template.bind({});
export const H5 = Template.bind({});
export const H6 = Template.bind({});

H1.args = {
  level: 'h1',
  children: 'Using cat rivalry to understand the <code>bind()</code> method',
};

H2.args = {
  level: 'h2',
  children: (
    <>
      Using cat rivalry to understand the <code>bind()</code> method
    </>
  ),
};

H3.args = {
  ...H1.args,
  level: 'h3',
};

H4.args = {
  ...H1.args,
  level: 'h4',
};

H5.args = {
  ...H1.args,
  level: 'h5',
};

H6.args = {
  ...H1.args,
  level: 'h6',
};
