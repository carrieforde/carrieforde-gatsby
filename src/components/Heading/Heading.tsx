import React from 'react';
import { useMemo } from 'react';
import { HeadingProps } from './Heading.interface';
import parse from 'html-react-parser';

const Heading: React.FC<HeadingProps> = ({ level, className, children }) => {
  const HeadingLevel = useMemo(() => level, []);
  const content = useMemo(
    () => (typeof children === 'string' ? parse(children) : children),
    []
  );

  return <HeadingLevel className={className}>{content}</HeadingLevel>;
};

export default Heading;
