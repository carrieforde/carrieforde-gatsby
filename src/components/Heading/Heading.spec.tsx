import { render } from "@testing-library/react";
import React from "react";
import { H1, H2 } from "./Heading.stories";

describe("Heading", () => {
  it("should render the Heading component as an h1", () => {
    const { container } = render(<H1 {...H1.args} />);
    expect(container).toMatchSnapshot();
  });

  it("should render the Heading component as an h2", () => {
    const { container } = render(<H2 {...H2.args} />);
    expect(container).toMatchSnapshot();
  });
});
