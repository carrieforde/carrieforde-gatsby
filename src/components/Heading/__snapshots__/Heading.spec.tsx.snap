// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render the Heading component as an h1 1`] = `
<div>
  <h1>
    Using cat rivalry to understand the 
    <code>
      bind()
    </code>
     method
  </h1>
</div>
`;

exports[`Heading should render the Heading component as an h2 1`] = `
<div>
  <h2>
    Using cat rivalry to understand the 
    <code>
      bind()
    </code>
     method
  </h2>
</div>
`;
